src/Resources/config/config.yml

```diff
@@ -25,6 +25,13 @@ winzou_state_machine:
                     on: ['complete']
                     do: ['@sylius_invoicing_plugin.event_producer.order_payment_paid', '__invoke']
                     args: ['object']
+    sylius_order_checkout:
+        callbacks:
+            after:
+                sylius_invoicing_plugin_order_placed_producer:
+                    on: ['complete']
+                    do: ['@sylius_invoicing_plugin.event_producer.order_placed', '__invoke']
+                    args: ['object']
```

Comment:
I wonder if it wouldn't be better to put this logic on the callback after `create` transition on `sylius_order` state machine 🤔

Reply:
ping @Prometee

Reply:
I'm not sure if I correctly understand you, is there a problem with moving the callback to `sylius_order` state machine, in this implementation or in both cases? What is more, as you probably suggested changing the moment of generating an invoice PDF, IMO it is not possible, it should be generated during invoice generation after the order is placed, not payment paid.

Reply:
During the `@sylius_invoicing_plugin.event_producer.order_placed'->__invoke()` the invoice creation is dispatched and cause the error I mentioned. To solve this error we can :

1.  remove the dispatch and simply call the service creating the `Invoice` with a new method signature for the class 
    `Sylius\InvoicingPlugin\Creator\InvoiceCreator`

    From :
    ```php
    public function __invoke(string $orderNumber, \DateTimeInterface $dateTime): void
    ```

    To:
    ```php
    public function __invoke(OrderInterface $order, \DateTimeInterface $dateTime): void
    // Maybe the argument `$dateTime` can be removed and we can use `$order->getCheckoutCompletedAt()` instead
    ```
2. Move the  PDF creation to another moment like when the PDF is sent `payment_state === 'paid'`

Reply:
From my point of view the `Invoice` data stored in the database are the immutable data, the PDF is a view of this data.